framework/test/unit/engine/legacy/endpoint.spec.ts

```diff
@@ -13,12 +13,62 @@
  * Removal or modification of this copyright notice is prohibited.
  */
 
+import { InMemoryDatabase } from '@liskhq/lisk-db';
+import { codec } from '@liskhq/lisk-codec';
+import { LegacyEndpoint } from '../../../../src/engine/legacy/endpoint';
+import { blockFixtures } from './fixtures';
+import { blockSchemaV2, blockHeaderSchemaV2 } from '../../../../src/engine/legacy/schemas';
+import { LegacyBlockJSON } from '../../../../src/engine/legacy/types';
+
+const bufferToHex = (b: Buffer) => Buffer.from(b).toString('hex');
+
 describe('Legacy endpoint', () => {
-	describe('getBlockByID', () => {
-		it.todo('test behaviors');
+	const legacyEndpoint = new LegacyEndpoint({ db: new InMemoryDatabase() as any });
+	const encodedBlock = codec.encode(blockSchemaV2, {
+		header: codec.encode(blockHeaderSchemaV2, blockFixtures[0].header),
+		transactions: blockFixtures[0].transactions,
 	});
 
-	describe('getBlockByHeight', () => {
-		it.todo('test behaviors');
+	legacyEndpoint.storage.getBlockByID = jest.fn().mockResolvedValue(encodedBlock);
+	legacyEndpoint.storage.getBlockByHeight = jest.fn().mockResolvedValue(encodedBlock);
+
+	describe('LegacyEndpoint', () => {
+		const matchExpectations = (block: LegacyBlockJSON) => {
```

Comment:
a suggestion on naming
```suggestion
		const matchBlockExpectations = (block: LegacyBlockJSON) => {
```